src/scriptdev2/scripts/kalimdor/darkshore.cpp

```diff
@@ -687,50 +688,35 @@ struct npc_rabid_bearAI : public ScriptedAI
     {
-        if (m_uiCheckTimer && m_creature->isInCombat())
+        if (pCaster->GetTypeId() == TYPEID_PLAYER && pSpell->Id == SPELL_BEAR_CAPTURED)
         {
-            if (m_uiCheckTimer <= uiDiff)
+            m_creature->RemoveAllAurasOnEvade();
+            m_creature->DeleteThreatList();
+            m_creature->CombatStop(true);
+            m_creature->SetLootRecipient(nullptr);
+            Reset();
+            // Update Entry and start following player
+            m_creature->UpdateEntry(NPC_CAPTURED_RABID_THISTLE_BEAR);
+            // get player
+            if (((Player*)pCaster)->GetQuestStatus(QUEST_PLAGUED_LANDS) == QUEST_STATUS_INCOMPLETE)
             {
-                // Trap nearby?
-                if (GameObject* pTrap = GetClosestGameObjectWithEntry(m_creature, GO_NIGHT_ELVEN_BEAR_TRAP, 0.5f))
-                {
-                    // Despawn trap
-                    pTrap->Use(m_creature);
-                    // "Evade"
-                    m_creature->RemoveAllAurasOnEvade();
-                    m_creature->DeleteThreatList();
-                    m_creature->CombatStop(true);
-                    m_creature->SetLootRecipient(nullptr);
-                    Reset();
-                    // Update Entry and start following player
-                    m_creature->UpdateEntry(NPC_CAPTURED_RABID_THISTLE_BEAR);
-                    // get player
-                    Unit* pTrapOwner = pTrap->GetOwner();
-                    if (pTrapOwner && pTrapOwner->GetTypeId() == TYPEID_PLAYER &&
-                            ((Player*)pTrapOwner)->GetQuestStatus(QUEST_PLAGUED_LANDS) == QUEST_STATUS_INCOMPLETE)
-                    {
-                        ((Player*)pTrapOwner)->KilledMonsterCredit(m_creature->GetEntry(), m_creature->GetObjectGuid());
-                        m_creature->GetMotionMaster()->MoveFollow(pTrapOwner, PET_FOLLOW_DIST, PET_FOLLOW_ANGLE);
-                    }
-                    else                                // Something unexpected happened
-                        m_creature->ForcedDespawn(1000);
-
-                    // No need to check any more
-                    m_uiCheckTimer = 0;
-                    // Despawn after a while (delay guesswork)
-                    m_uiDespawnTimer = 3 * MINUTE * IN_MILLISECONDS;
-
-                    return;
-                }
-                else
-                    m_uiCheckTimer = 1000;
+                ((Player*)pCaster)->KilledMonsterCredit(m_creature->GetEntry(), m_creature->GetObjectGuid());
+                m_creature->GetMotionMaster()->MoveFollow(pCaster, PET_FOLLOW_DIST, PET_FOLLOW_ANGLE);
             }
-            else
-                m_uiCheckTimer -= uiDiff;
+            else                                // Something unexpected happened
+                m_creature->ForcedDespawn(1000);
+
+            // No need to check any more
+            m_uiCheckTimer = 0;
+            // Despawn after a while (delay guesswork)
```

Comment:
what's the name of this quest, maybe I can help find out what the despawn time is.